R/FilteredData-utils.R

```diff
@@ -29,7 +29,10 @@ init_filtered_data <- function(x, join_keys = teal.data::join_keys(), code, chec
       "init_filtered_data(check = 'No longer supported')"
     )
   }
-  FilteredData$new(x, join_keys = join_keys)
+  FilteredData$new(
+    Filter(function(obj) inherits(obj, c("data.frame", "MultiAssayExperiment")), x),
+    join_keys = join_keys
+  )
```

Comment:
I echo @chlebowa comment [here](https://github.com/insightsengineering/teal.slice/pull/624#pullrequestreview-2385205495). I think this should be delegated lower into the dataset constructor. Conceptually - a `function` (or anything like that) cannot become a filtered dataset.